subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/incremental/IncrementalCompilerDecorator.java

```diff
@@ -72,8 +72,10 @@ public IncrementalCompilerDecorator(JarClasspathSnapshotMaker jarClasspathSnapsh
             return cleaningCompiler;
         }
         if (!annotationProcessorPath.isEmpty()) {
```

Comment:
This needs to be replaced with a test that checks if the AP path is empty *or* all processors are incremental.